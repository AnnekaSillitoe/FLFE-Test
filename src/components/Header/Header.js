import React from "react";
import PropTypes from "prop-types";

const Header = ({ onClick }) => {
  return (
    <header className="header">
      <button className="header-admin-button" onClick={onClick}>
        <div className="icon icon-pull" />
      </button>
      <h1 className="header-title">FutureLearn</h1>
    </header>
  );
};

Header.propTypes = {
  onClick: PropTypes.func.isRequired
};

export default Header;
