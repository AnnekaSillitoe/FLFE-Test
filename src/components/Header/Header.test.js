import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";

const mockEvent = jest.fn();
const props = {
  onClick: mockEvent,
};

test("Header renders correctly", () => {
  const header = shallow(<Header {...props} />);
  expect(header).toMatchSnapshot();
});

test("When clicking the button, it triggers the onClick", () => {
  const header = shallow(<Header {...props} />);
  const button = header.find("button");

  button.simulate("click");

  expect(mockEvent.mock.calls.length).toEqual(1);
});
