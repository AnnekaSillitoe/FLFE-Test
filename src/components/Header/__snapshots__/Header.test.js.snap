// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="header-admin-button"
          onClick={[MockFunction]}
        >
          <div
            className="icon icon-pull"
          />
        </button>,
        <h1
          className="header-title"
        >
          FutureLearn
        </h1>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="icon icon-pull"
          />,
          "className": "header-admin-button",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "icon icon-pull",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FutureLearn",
          "className": "header-title",
        },
        "ref": null,
        "rendered": "FutureLearn",
        "type": "h1",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="header-admin-button"
            onClick={[MockFunction]}
          >
            <div
              className="icon icon-pull"
            />
          </button>,
          <h1
            className="header-title"
          >
            FutureLearn
          </h1>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="icon icon-pull"
            />,
            "className": "header-admin-button",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "icon icon-pull",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FutureLearn",
            "className": "header-title",
          },
          "ref": null,
          "rendered": "FutureLearn",
          "type": "h1",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
